lib/contentful_middleman/helpers.rb

```diff
@@ -7,7 +7,8 @@ def contentful_instances
     end
 
     def localize_entry(entry, locale, fallback_locale='en-US')
-      localized_entry = {}
+      localized_entry = entry.class.new
+      localized_entry = {} unless localized_entry.is_a? ::Hash
```

Comment:
This can be changed to

```ruby
localized_entry = ::Thor::CoreExt::HashWithIndifferentAccess.new
```

As it is a superclass of `::Hash`, there should be absolutely no difference with the current implementation.

Just make sure to add the explicit require at the top of the file.